File: beepi/py/fltr_val_intrvl.py
#!/usr/bin/python

#############################
# @author: vladimir kulyukin
#############################
import re
import sys

val_index  = int(sys.argv[1])
lower_bound = int(sys.argv[2])
upper_bound = int(sys.argv[3])
for line in sys.stdin.readlines():
    match = re.match(r'\S+', line)
    if match != None:
        val = int(re.split(r'\t', line)[val_index])
        if lower_bound <= val and val <= upper_bound:
            sys.stdout.write(line)

File: beepi/pl/fltr_val_intrvl.pl
#!/usr/bin/perl

################################################################
## @author: vladimir kulyukin
################################################################    
use strict;
use warnings;

my $VAL_INDEX         = shift();
my $LOWER_BOUND = shift();
my $UPPER_BOUND  = shift();
while ( <> ) {
    if ( $_ =~ /\S+/ ) {
	my @splits = split(/\t/, $_);
	my $val = $splits[$VAL_INDEX];
	print $_ if ( $val >= $LOWER_BOUND && $val <= $UPPER_BOUND );
    }
}
